From b2ddc13858bb287cd19e8e16b49c756d33447404 Mon Sep 17 00:00:00 2001
From: Xinyu Chen <xinyu.chen@freescale.com>
Date: Fri, 29 Aug 2008 10:53:16 +0800
Subject: [PATCH] ENGR00083026 DMA memory is not enough for some H264 clips on MX37

Increase the CONSISTENT_DMA_SIZE to 32MB to make sure
the VM has region for maximum 32MB dma_alloc
Set MX37 default DMA zone size to 32MB

Signed-off-by: Xinyu Chen <xinyu.chen@freescale.com>
---
 arch/arm/configs/imx37_3stack_defconfig |    2 +-
 include/asm-arm/arch-mxc/memory.h       |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/configs/imx37_3stack_defconfig b/arch/arm/configs/imx37_3stack_defconfig
index fa5acc3..d2f6761 100644
--- a/arch/arm/configs/imx37_3stack_defconfig
+++ b/arch/arm/configs/imx37_3stack_defconfig
@@ -184,7 +184,7 @@ CONFIG_ARCH_MXC_HAS_NFC_V3_1=y
 #
 CONFIG_MXC_TZIC=y
 CONFIG_ARCH_HAS_EVTMON=y
-CONFIG_DMA_ZONE_SIZE=24
+CONFIG_DMA_ZONE_SIZE=32
 CONFIG_UTMI_MXC=y
 
 #
diff --git a/include/asm-arm/arch-mxc/memory.h b/include/asm-arm/arch-mxc/memory.h
index 243c606..c1d648c 100644
--- a/include/asm-arm/arch-mxc/memory.h
+++ b/include/asm-arm/arch-mxc/memory.h
@@ -32,7 +32,7 @@
 #endif
 
 /* Size of contiguous memory for DMA and other h/w blocks */
-#define CONSISTENT_DMA_SIZE	(14 * SZ_1M)
+#define CONSISTENT_DMA_SIZE	(32 * SZ_1M)
 
 #ifndef __ASSEMBLY__
 
-- 
1.5.4.4

